subworkflows/local/dada2_preprocessing.nf

```diff
@@ -74,20 +74,52 @@ workflow DADA2_PREPROCESSING {
     DADA2_FILTNTRIM ( ch_trimmed_reads.dump(tag: 'into_filtntrim')  )
     ch_versions_dada2_preprocessing = ch_versions_dada2_preprocessing.mix(DADA2_FILTNTRIM.out.versions.first())
 
+    //Filter empty files
+    DADA2_FILTNTRIM.out.reads_logs_args
+        .branch {
+            failed: it[0].single_end ? it[1].countFastq() < params.min_read_counts : it[1][0].countFastq() < params.min_read_counts || it[1][1].countFastq() < params.min_read_counts
+            passed: true
+        }
+        .set { ch_dada2_filtntrim_results }
+    ch_dada2_filtntrim_results.passed.set { ch_dada2_filtntrim_results_passed }
+    ch_dada2_filtntrim_results.failed
+        .map { meta, reads, logs, args -> [ meta.id ] }
+        .collect()
+        .subscribe {
+            samples = it.join("\n")
+            if (params.ignore_failed_filtering) {
+                log.warn "The following samples had too few reads (<$params.min_read_counts) after quality filtering with DADA2:\n$samples\nIgnoring failed samples and continue!\n"
+            } else {
+                error("The following samples had too few reads (<$params.min_read_counts) after quality filtering with DADA2:\n$samples\nPlease check whether the correct primer sequences for trimming were supplied. Ignore that samples using `--ignore_failed_filtering` or adjust the threshold with `--min_read_counts`.")
+            }
+        }
+
+    // Break apart the reads and logs so that only the samples
+    // which pass filtering are retained
+    ch_dada2_filtntrim_results_passed
+        .map{ it -> [it[0], it[1]] }
```

Comment:
That seems more verbose, would it work? If yes the other two might benefit as well, I like more verbose
```suggestion
        .map{ meta, reads, logs, args -> [meta, reads] }
```